syntax/expr_binary_test.go

```diff
@@ -19,13 +19,19 @@ func TestRelationCall(t *testing.T) {
 
 func TestOpsAddArrow(t *testing.T) {
 	t.Parallel()
+	// Tuples
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 2, c: 3, d: 4)`, `(a: 1, b: 2) +> (c: 3, d: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 3, c: 4)`, `(a: 1, b: 2) +> (b: 3, c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: 3, c: 4)`, `(a: 1, b: (c: 2)) +> (b: 3, c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 4), c: 4)`, `(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)`)
+	AssertCodesEvalToSameValue(t, `(a: (b: 1), b: (c: 4))`, `(a: 1, b: (c: 2)) +> (a: (b: 1), b: (c: 4))`)
 
-	AssertCodesEvalToSameValue(t, `(a: 1, b: 2) +> (c: 3, d: 4)`, `(a: 1, b: 2, c: 3, d: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: 2) +> (b: 3, c: 4)`, `(a: 1, b: 3, c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (b: 3, c: 4)`, `(a: 1, b: 3, c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (b: (c: 4), c: 4)`, `(a: 1, b: (c: 4), c: 4)`)
-	AssertCodesEvalToSameValue(t, `(a: 1, b: (c: 2)) +> (a: (b: 1), b: (c: 4))`, `(a: (b: 1), b: (c: 4))`)
+	// Dicts
+	AssertCodesEvalToSameValue(t, `{'a': 1, 'b': 3, 'd': 4}`, `{'a': 1, 'b': 2} +> {'b': 3, 'd': 4}`)
+	AssertCodesEvalToSameValue(t, `{'a': 1, 'b': 3, 'd': 4}`, `{'a': 1, 'b': {'a': 2}} +> {'b': 3, 'd': 4}`)
+	AssertCodesEvalToSameValue(t, `{'a': {'c': 2}}`, `{'a': {'b': 1}} +> {'a': {'c': 2}}`)
 
+	//
```

Comment:
These can be a separate case as well

Reply:
Also, maybe add a test for lists and strings? They're more dict-like than sets and ints